tests/data/test_stub_generation/function_module.py

```diff
@@ -0,0 +1,60 @@
+class A:
+    ...
+
+
+class B:
+    def __init__(self, init_param): ...
+
+    def class_method(self, a: A) -> A: ...
+
+    @staticmethod
+    def static_class_method() -> None: ...
+
+
+def _private(a): ...
+
+
+def public_no_params_no_result(): ...
+
+
+def params(
+    i: int,
+    union: int | bool,
+    lst: list[int],
+    obj: A,
+): ...
+
+
+def illegal_params(
```

Comment:
Why are these parameters illegal? Tuples, yes, the faulty default value, yes, but not the others.